component/prometheus/remotewrite/types.go

```diff
@@ -108,84 +182,33 @@ func (o *WALOptions) UnmarshalRiver(f func(interface{}) error) error {
 }
 
-// BasicAuthConfig is the prometheus.remote_write's configuration for
-// authenticating against the remote system when sending metrics.
-type BasicAuthConfig struct {
-	Username     string            `river:"username,attr"`
-	Password     rivertypes.Secret `river:"password,attr,optional"`
-	PasswordFile string            `river:"password_file,attr,optional"`
-}
-
-// UnmarshalRiver allows injecting of default values
-func (r *Config) UnmarshalRiver(f func(v interface{}) error) error {
-	*r = Config{
-		SendExemplars: true,
-	}
-
-	type arguments Config
-	return f((*arguments)(r))
-}
-
-// UnmarshalRiver allows injecting of default values
-func (r *QueueConfig) UnmarshalRiver(f func(v interface{}) error) error {
-	*r = DefaultQueueConfig
-
-	type arguments QueueConfig
-	return f((*arguments)(r))
-}
-
-func convertConfigs(cfg RemoteConfig) (*config.Config, error) {
+func convertConfigs(cfg Arguments) (*config.Config, error) {
 	var rwConfigs []*config.RemoteWriteConfig
-	for _, rw := range cfg.RemoteWrite {
+	for _, rw := range cfg.Endpoints {
 		parsedURL, err := url.Parse(rw.URL)
 		if err != nil {
 			return nil, fmt.Errorf("cannot parse remote_write url %q: %w", rw.URL, err)
 		}
 
-		rwc := &config.RemoteWriteConfig{
-			Name:          rw.Name,
-			URL:           &common.URL{URL: parsedURL},
-			RemoteTimeout: model.Duration(30 * time.Second),
-			QueueConfig:   config.DefaultQueueConfig,
-			MetadataConfig: config.MetadataConfig{
-				Send: false,
-			},
-			HTTPClientConfig: common.DefaultHTTPClientConfig,
-		}
-
-		if rw.BasicAuth != nil {
-			rwc.HTTPClientConfig.BasicAuth = &common.BasicAuth{
-				Username:     rw.BasicAuth.Username,
-				Password:     common.Secret(rw.BasicAuth.Password),
-				PasswordFile: rw.BasicAuth.PasswordFile,
-			}
-		}
-
-		if rw.QueueConfig != nil {
-			rwc.QueueConfig = config.QueueConfig{
-				Capacity:          rw.QueueConfig.Capacity,
-				MaxShards:         rw.QueueConfig.MaxShards,
-				MinShards:         rw.QueueConfig.MinShards,
-				MaxSamplesPerSend: rw.QueueConfig.MaxSamplesPerSend,
-				BatchSendDeadline: model.Duration(rw.QueueConfig.BatchSendDeadline),
-				MinBackoff:        model.Duration(rw.QueueConfig.MinBackoff),
-				MaxBackoff:        model.Duration(rw.QueueConfig.MaxBackoff),
-				RetryOnRateLimit:  rw.QueueConfig.RetryOn429,
-			}
-		}
-
-		if rw.HTTPClientConfig != nil {
-			rwc.HTTPClientConfig = *rw.HTTPClientConfig.Convert()
-		}
-
-		rwc.SendExemplars = rw.SendExemplars
-
-		rwConfigs = append(rwConfigs, rwc)
+		rwConfigs = append(rwConfigs, &config.RemoteWriteConfig{
+			URL:                 &common.URL{URL: parsedURL},
+			RemoteTimeout:       model.Duration(rw.RemoteTimeout),
+			Headers:             rw.Headers,
+			WriteRelabelConfigs: nil, // WriteRelabelConfigs are currently not supported
+			Name:                rw.Name,
+			SendExemplars:       rw.SendExemplars,
+
+			HTTPClientConfig: *rw.HTTPClientConfig.Convert(),
```

Comment:
WDYT about reusing [this](https://github.com/grafana/agent/blob/917b8f3eaedcef34fa8316a416ae3fc4c07286d2/component/prometheus/scrape/types.go#L52) validation method before applying the HTTP client config here? 

We use it in prometheus.scrape where we fail the component creation if conflicting configuration is passed eg. defining both basic_auth _and_ oauth2. I picked it up from [prometheus/common](https://github.com/prometheus/common/blob/dfbc25bd00225c70aca0d94c3c4bb7744f28ace0/config/http_config.go#L278-L338) where it's called in their UnmarshalYAML/UnmarshalJSON methods.


The validation could even go in the struct's Convert method, (although it's a little awkward) or the UnmarshalRiver itself.

Reply:
Good idea. I made Validate a method on the struct and called it during unmarshal.

Reply:
@tpaschalis can you take another look to see if what I did looks ok to you?

Reply:
Yeah, works fine for both components during initial load and reloaded configuration. Thanks!